### PR TITLE
Scaffold a new controller using kubebuilder

### PR DESCRIPTION
**Pull request by mfaizanse**
**Description**

Changes proposed in this pull request:

- Scaffold a new controller using kubebuilder
- [Checklist](https://github.com/kyma-project/eventing-manager/issues/4#issuecomment-1571904737)

**Related issue(s)**

- https://github.com/kyma-project/eventing-manager/issues/4

### CODE REVIEW

File: api/v1alpha1/eventing_types.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+// EventingStatus defines the observed state of Eventing
+type EventingStatus struct {
+	State      string             `json:"state"`
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// Eventing is the Schema for the eventings API
```
Comment (friedrichwilken) [inline api/v1alpha1/eventing_types.go:59]: this way it becomes clearer that the decorators belong to `Eventing`.

```suggestion
// Eventing is the Schema for the eventing API.
//+kubebuilder:object:root=true
//+kubebuilder:subresource:status
```

File: api/v1alpha1/eventing_types.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   EventingSpec   `json:"spec,omitempty"`
+	Status EventingStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// EventingList contains a list of Eventing
```
Comment (friedrichwilken) [inline api/v1alpha1/eventing_types.go:70]:
```suggestion

// EventingList contains a list of Eventing.
//+kubebuilder:object:root=true
```

File: api/v1alpha1/eventing_types.go [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
```
Comment (friedrichwilken) [inline api/v1alpha1/eventing_types.go:25]: begone
```suggestion
```

File: go.mod [hunk 4]
```diff
@@ -0,0 +1,70 @@
+module github.com/kyma-project/eventing-manager
+
+go 1.19
```
Comment (friedrichwilken) [inline go.mod:3]:
```suggestion
go 1.20
```

**Review by friedrichwilken — CHANGES_REQUESTED** [review]

File: api/v1alpha1/eventing_types.go (hunk 2 above)
Comment (mfaizanse) [reply api/v1alpha1/eventing_types.go]: updated

File: README.md [hunk 5]
```diff
@@ -1,39 +1,229 @@
...
+# Eventing Manager
+Manages the lifecycle of Eventing resources.
 
-# {Project Title} (mandatory)
+## Description
+It is a standard Kubernetes operator which observes the state of Eventing resources and reconciles its state according to desired state.
 
-> Modify the title and insert the name of your project. Use Heading 1 (H1).
+## Getting Started
+You’ll need a Kubernetes cluster to run against. You can use [KIND](https://sigs.k8s.io/kind) to get a local cluster for testing, or run against a remote cluster.
```
[the remark this answers is not in the document]
Comment (mfaizanse) [reply README.md]: updated

**Review by friedrichwilken — APPROVED** [review]

**Pull request merged**